src/app.tsx

```diff
@@ -67,7 +67,7 @@ export const Application = () => {
 
     const onFilterChange = (_event: React.FormEvent<HTMLInputElement>, value: string) => setCurrentFilter(value);
     const currentPath = decodeURIComponent(options.path?.toString() || "");
-    const path = currentPath?.split("/");
+    const path = React.useMemo(() => currentPath?.split("/"), [currentPath]);
```

Comment:
This feels a bit overdoing it, splitting a path can not be that slow. Honestly we should moved `selected` somewhere else. But the issue is that we created the sidepanel, so likely we should just throw that into a new Component with the SidePanel.